akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingGetStatsSpec.scala

```diff
@@ -175,29 +173,32 @@ abstract class ClusterShardingGetStatsSpec extends MultiNodeSpec(ClusterSharding
           regions.keys.forall(_.hasGlobalScope) should be(true)
         }
       }
-
       enterBarrier("got shard state")
-
+      system.log.info("got shard state")
     }
 
     "return stats after a node leaves" in {
-      runOn(first) {
+      runOn(controller) {
         Cluster(system).leave(node(third).address)
       }
 
-      runOn(third) {
-        watch(shardActor)
-        expectTerminated(shardActor, 15.seconds)
+      runOn(controller, first, second) {
+        within(10.seconds) {
```

Comment:
it might take longer than 10 seconds for a node to leave completely, failure detector must trigger